File: 1014-k-closest-points-to-origin/k-closest-points-to-origin.cpp
class Solution {
public:
    vector<vector<int>> kClosest(vector<vector<int>>& points, int k) {
        priority_queue<vector<int>> q;
        vector<vector<int>> result;

        for(auto& point: points){
            int x = point[0];
            int y = point[1];
            q.push({x * x + y * y, x, y});
            if(q.size() > k){
                q.pop();
            }
        }
        for(int i = 0; i < k; i++){
            vector<int> p = q.top();
            q.pop();
            result.push_back({p[1],p[2]});
        }
        return result;
    }
};

File: 1014-k-closest-points-to-origin/k-closest-points-to-origin.py
class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        heap = []

        for point in points:
            dist = -(point[0] * point[0] + point[1] * point[1])
            if len(heap) == k:
                heapq.heappushpop(heap, (dist, point[0], point[1]))
            else:
                heapq.heappush(heap,(dist, point[0], point[1]))
        
        return [(x,y) for (dist, x, y) in heap]

        